### PR TITLE
Add configurable JAX kernel block dimensions

### LINKED ISSUE #1436: Provide a way to specify block size for jax_kernel
### Description
The jax_kernel FFI path in `warp/_src/jax_experimental/ffi.py` hardcodes `block_dim=256` in the `wp_cuda_launch_kernel` call (line 437). There is no way to specify block_dim through the jax_kernel API.

Request: expose `block_dim` as a parameter to `jax_kernel()` and/or as a per-call override, analogous to how `wp.launch()` accepts it.

### Context

The hard-coded block size prevents tuning for kernels where the thread block is an important unit of assigning work. For the ALCHEMI toolkit, we'd like to process one atom per block so that tile-based reduction API can then be used. Using a smaller block size is desirable because a block size of 256 results in too less work per thread.

### PR DESCRIPTION
**Pull request by shi-eric**
## Description

`wp.jax_kernel()` currently fixes CUDA launches at 256 threads per block, preventing block-size tuning and excluding tile kernels that require another cooperative width. This change adds a construction-time `block_dim` argument and consistently applies it to module preloading, direct FFI callbacks, forward launches, and generated adjoint launches. CPU execution continues to use one thread per block.

Closes #1436.

## Changes

- Add and validate the `block_dim` argument on `wp.jax_kernel()`.
- Include the selected block dimension in direct and differentiable wrapper cache identities.
- Load and launch the matching CUDA module variant while preserving the CPU block dimension of one.
- Apply the configured block dimension to both forward and adjoint launches when automatic differentiation is enabled.
- Document CUDA block dimensions, tile-kernel launch mapping, and output-dimension behavior.
- Add user-facing changelog coverage.

## Checklist

- [x] I am familiar with the [Contributing Guidelines](https://nvidia.github.io/warp/latest/project/contribution_guide.html).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.
- [x] [CHANGELOG.md](CHANGELOG.md) is updated for any user-facing changes under the `Unreleased` section.

## Validation summary

- Verified argument validation and normalization through `test_ffi_jax_kernel_block_dim_validation`.
- Verified direct JAX FFI tile reductions with 64- and 128-thread CUDA blocks on both available CUDA devices.
- Mutation-checked the direct-wrapper cache coverage by removing `block_dim` from the cache key, observing the expected 128-thread failures, restoring the key, and confirming the tests pass.
- Verified forward and adjoint block-dimension propagation on CPU and both available CUDA devices through `test_ffi_jax_kernel_block_dim_autodiff`.
- Ran the repository pre-commit hooks for the changed test module.

## New feature / enhancement

```python
import jax
import warp as wp

TILE_THREADS = 64

@wp.kernel
def block_sum(output: wp.array[int]):
    i = wp.tid()
    output[i] = 0
    values = wp.tile(i)
    sums = wp.tile_sum(values)
    wp.tile_store(output, sums, offset=i)

jax_block_sum = wp.jax_kernel(
    block_sum,
    launch_dims=256,
    output_dims=256,
    block_dim=TILE_THREADS,
)

(output,) = jax.jit(jax_block_sum)()
```

## Summary by CodeRabbit

* **New Features**
  * Added optional `block_dim` to `wp.jax_kernel()` to control CUDA thread-block size, including tile-kernel execution.
  * `block_dim` is applied to both forward execution and autodiff/adjoint launches; different values use separate cached wrappers.
  * The returned wrapper exposes `block_dim`, with defaults matching existing CUDA behavior.
* **Documentation**
  * Updated the JAX interoperability guide with “CUDA Block Dimensions and Tile Kernels,” including launch/output-shape conventions.
* **Tests**
  * Added end-to-end JAX FFI tests for `block_dim` (tiled behavior, gradients, validation, and CPU/CUDA preload rules).

### CODE REVIEW

File: warp/_src/jax/ffi.py [hunk 1]
```diff
@@ -1338,6 +1349,16 @@ def jax_kernel(
...
 
+    if block_dim is not None:
+        if isinstance(block_dim, bool):
+            raise TypeError("jax_kernel(): block_dim must be an integer or None.")
+        try:
+            block_dim = operator.index(block_dim)
+        except TypeError:
+            raise TypeError("jax_kernel(): block_dim must be an integer or None.") from None
+        if block_dim <= 0:
+            raise ValueError("jax_kernel(): block_dim must be positive.")
```
[the remark this answers is not in the document]
Comment (shi-eric) [reply warp/_src/jax/ffi.py:1362]: @greptileai Warp intentionally defers CUDA block-size constraints to the CUDA launch path. This matches `wp.launch()`, which allows the driver to validate both the device limit and kernel-specific constraints such as `launch_bounds`. A hard-coded 1024 check here would be JAX-specific and incomplete because a kernel’s effective maximum may be lower.
